v1/ao/internal/reporter/oboe.go

```diff
@@ -296,28 +302,28 @@ func updateSetting(sType int32, layer string, flags []byte, value int64, ttl int
 	ns.ttl = ttl
 	ns.layer = layer
 
-	ns.bucket.capacity = parseFloat64(args, kvBucketCapacity, 0)
-	ns.bucket.ratePerSec = parseFloat64(args, kvBucketRate, 0)
+	rate := parseFloat64(args, kvBucketRate, 0)
+	capacity := parseFloat64(args, kvBucketCapacity, 0)
+	ns.bucket.setRateCap(rate, capacity)
 
 	key := oboeSettingKey{
 		sType: settingType(sType),
 		layer: layer,
 	}
 
 	globalSettingsCfg.lock.Lock()
-	if s, ok := globalSettingsCfg.settings[key]; ok {
-		ns.bucket.setAvail(s.bucket.avail())
-	}
```

Comment:
Removing these lines might cause available token reset to zero every time a new settings arrive?

Reply:
ah make sense. It's fine then :+1: